### ISSUE TITLE
training divergence

### ISSUE THREAD

**Issue by sven-x** [HorizonRobotics/Sparse4D#4]
Hi @linxuewu , I tried the given sparse4Dv3 R50 config with 8 GPUs, the training diverges with the following log attached. Could you please check it? Thanks.
[log.json](https://github.com/HorizonRobotics/Sparse4D/files/14021361/log.json)

**Comment by linxuewu** [comment]
The divergence speed is too fast, there should be a significant bug. It is recommended to run inference with the [ckpt](https://github.com/HorizonRobotics/Sparse4D/releases/download/v3.0/sparse4dv3_r50.pth) I provided first, aligning the metrics.

**Comment by linxuewu** [comment]
I tried it, and it indeed diverges. I'll locate the issue. It seems there was a problem during the code open-source merge.

**Comment by linxuewu** [comment]
I fix a bug in data transform, which will get incorrect "focal". Please pull the latest code and try again.

**Comment by ASONG0506** [comment]
[saprse4d_20240124_101358.log](https://github.com/HorizonRobotics/Sparse4D/files/14045759/saprse4d_20240124_101358.log)
hello @linxuewu ,I pulled the newest version code, and train on nuscenes dataset, I modified the batchsize to 2 with 4gpus, but the result is nan.
Could please help me to check what's the problem here? thank you.

**Comment by ASONG0506** [comment]
[saprse4d_20240124_114437.log](https://github.com/HorizonRobotics/Sparse4D/files/14045823/saprse4d_20240124_114437.log)
hello @linxuewu here is another training log with 4gpus and batchsize=6, the training was interpreted with some error

**Comment by linxuewu** [comment]
> [saprse4d_20240124_114437.log](https://github.com/HorizonRobotics/Sparse4D/files/14045823/saprse4d_20240124_114437.log) hello @linxuewu here is another training log with 4gpus and batchsize=6, the training was interpreted with some error

If number of GPUs reduces from 8 to 4, the learning rate also needs to be appropriately lowered. You can adjust the learning rate to 4e-4 or 3e-4 and try again.

**Comment by ASONG0506** [comment]
> > [saprse4d_20240124_114437.log](https://github.com/HorizonRobotics/Sparse4D/files/14045823/saprse4d_20240124_114437.log) hello @linxuewu here is another training log with 4gpus and batchsize=6, the training was interpreted with some error
> 
> If number of GPUs reduces from 8 to 4, the learning rate also needs to be appropriately lowered. You can adjust the learning rate to 4e-4 or 3e-4 and try again.

ok, thx, I'll try it

**Comment by ASONG0506** [comment]
> > [saprse4d_20240124_114437.log](https://github.com/HorizonRobotics/Sparse4D/files/14045823/saprse4d_20240124_114437.log) hello @linxuewu here is another training log with 4gpus and batchsize=6, the training was interpreted with some error
> 
> If number of GPUs reduces from 8 to 4, the learning rate also needs to be appropriately lowered. You can adjust the learning rate to 4e-4 or 3e-4 and try again.

it seems like my issue is solved
